processor/filterprocessor/factory.go

```diff
@@ -53,7 +54,15 @@ func createMetricsProcessor(
 	cfg component.Config,
 	nextConsumer consumer.Metrics,
 ) (processor.Metrics, error) {
-	fp, err := newFilterMetricProcessor(set.TelemetrySettings, cfg.(*Config))
+	oCfg := cfg.(*Config)
+	if oCfg.Metrics.Include != nil || oCfg.Metrics.Exclude != nil {
+		set.Logger.Warn(
+			"The metric `include` and `exclude` configuration options have been deprecated and will be removed in 0.74.0. Use `metric` instead.",
```

Comment:
Is there anything we can link here to help end-users understand how their old configuration should be rewritten in OTTL statements? The docs on how to use `metric` (same for `log_record` and `traces.span`) tell you to use OTTL, but if you are completely new to OTTL I think it would help to have a bit more hand-holding here (maybe a link to an issue where we write some migration instructions?).

Reply:
> maybe a link to an issue where we write some migration instructions?

I can add documentation on how to convert each signal's existing configuration options into OTTL.  I could do that right in the README instead of an issue, what do you think?

Reply:
Is mechanical conversion possible?  If so, should we provide a [`confmap.Converter`](https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/converter.go#L23) implementation that can automatically handle the conversion?

Reply:
@Aneurysm9 I'm not familiar with that interface yet, but it is _possible_ that a conversion could be written. I don't think it would be trivial, but hopefully reusable for other components that implement `internal/filter` config structs.

If we go that route, I would vote it not be a permanent solution.  I would still want to eventually deprecate the old configuration style and instead support declarative syntax via an [OTTL-specific declarative format](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11852) long term.

Reply:
For this PR, I am fine with anything that an inexperienced user can grasp easily without having to search through the docs. I think the Converter idea sounds nice, but I don't think we should block this PR on that work

Reply:
I think if the converter will not be permanent then the value of developing one may be moot. I could see the argument of a converter doing more harm than good also. Users may end up relying on the converter that will end up being removed. It may be better to invest in a migration guide and ensure that OTTL documentation is ready for more users/components to rely on.